whois-nrtm4-client/src/main/java/net/ripe/db/nrtm4/client/importer/SnapshotImporter.java

```diff
@@ -0,0 +1,170 @@
+        // parallelise the process
+        String[] snapshotRecords;
+        byte[] payload;
+        try {
+            payload = nrtmRestClient.getSnapshotFile(snapshot.getUrl());
+            snapshotRecords = getSnapshotRecords(payload);
+        } catch (IOException e){
+            LOGGER.error("No able to decompress snapshot", e);
+            return;
+        }
+
+        if (snapshotRecords == null){
+            LOGGER.error("This cannot happen. UNF has a non-existing snapshot");
+            return;
+        }
+
+        final JSONObject jsonObject = new JSONObject(snapshotRecords[0]);
+        final int snapshotVersion = jsonObject.getInt("version");
+        final String snapshotSessionId = jsonObject.getString("session_id");
+
+        if (!snapshot.getHash().equals(calculateSha256(payload))){
+            LOGGER.error("Snapshot hash doesn't match, skipping import");
+            return;
+        }
+
+        if (!snapshotSessionId.equals(updateNotificationFile.getSessionID())){
+            // TODO: [MH] if the service is wrong for any reason...we have here a non-ending loop, we need to
+            //  call initialize X number of times and return error to avoid this situation?
+            LOGGER.error("The session is not the same in the UNF and snapshot");
+            //initializeNRTMClientForSource(source, updateNotificationFile);
+            return;
+        }
+
+        printProgress(new Timer());
+        Arrays.stream(snapshotRecords).skip(1)
+                .parallel()
+                .forEach(record -> {
+                    try {
+                        processObject(record);
+                        processedCount.incrementAndGet();
+                    } catch (JsonProcessingException e) {
+                        LOGGER.error("Unable to process record", e);
+                    }
+                });
+
+        nrtm4ClientMirrorDao.saveSnapshotFileVersion(source, snapshotVersion, snapshotSessionId);
+        LOGGER.info("Loading snapshot file took {} for source {}", stopwatch.elapsed().toMillis(), source);
+    }
+
+    private void processObject(final String record) throws JsonProcessingException {
+        final MirrorRpslObject mirrorRpslObject = new ObjectMapper().readValue(record, MirrorRpslObject.class);
+        nrtm4ClientMirrorDao.persistRpslObject(mirrorRpslObject.getObject());
+    }
+
+    private void printProgress(final Timer timer) {
+        timer.schedule(new TimerTask() {
+            @Override
+            public void run() {
+                LOGGER.info("Records processed so far: {}", processedCount.get());
+            }
+        }, 0, 10000);
+    }
+
+    private static String[] getSnapshotRecords(byte[] compressed) throws IOException {
+        return StringUtils.split(decompress(compressed), RECORD_SEPARATOR);
+    }
+
+    private static String decompress(final byte[] compressed) throws IOException {
+        final int BUFFER_SIZE = 4096;
```

Comment:
If you intend `BUFFER_SIZE` to be a static field then declare it so at the top of the class